src/Common/Notifications/Controller.php

```diff
@@ -0,0 +1,128 @@
+<?php
+/**
+ * Controller for In-App Notifications.
+ *
+ * @since   TBD
+ *
+ * @package TEC\Common\Notifications
+ */
+
+namespace TEC\Common\Notifications;
+
+use TEC\Common\Contracts\Provider\Controller as Controller_Contract;
+
+/**
+ * Class Controller
+ *
+ * @since   TBD
+
+ * @package TEC\Common\Notifications
+ */
+class Controller extends Controller_Contract {
+
+	/**
+	 * Registers actions and filters.
+	 *
+	 * @since TBD
+	 */
+	public function do_register(): void {
+		$this->add_actions();
+	}
+
+	/**
+	 * Unhooks actions and filters.
+	 */
+	public function unregister(): void {
+		$this->remove_actions();
+	}
+
+	/**
+	 * Add the action hooks.
+	 *
+	 * @since TBD
+	 */
+	public function add_actions() {
+		add_action( 'tribe_plugins_loaded', [ $this, 'boot' ], 50 );
```

Comment:
What i see that this hook is trying to - which is the only thing that needs to be done in a hook is register assets.

So lets do that instead of booting. Also i did dig into when the boot method is called and its called when the whole app is being booted.

It goes like all the registration happens then when the container calls boot all the providers overloading the container call their boot method as well. Its not usefull in frameworks like WP, because iduring registration you can simply add a hook to execute later (when you want) a callback.